provider/coredns/coredns.go

```diff
@@ -106,16 +112,29 @@ func (c etcdClient) GetServices(ctx context.Context, prefix string) ([]*Service,
 	var svcs []*Service
 	bx := make(map[Service]bool)
 	for _, n := range r.Kvs {
-		svc := new(Service)
-		if err := json.Unmarshal(n.Value, svc); err != nil {
-			return nil, fmt.Errorf("%s: %w", n.Key, err)
+		svc, err := c.unmarshalService(n)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
```suggestion
		}
		if c.strictlyOwned && svc.OwnedBy != c.ownerID {
			continue
		}
```

You could just do Early ownership check: skip records not owned by this instance when strictlyOwned.